### PR TITLE
Svg/percentages

### PR DESCRIPTION
**Pull request by dpsutton**
## DRAFT

Draft until the main PR is merged https://github.com/metabase/metabase/pull/15781

Include percentages on categorical donut charts

#### What it looks like
image

To reproduce, create a question that is a good candidate for a donut chart: two columns one of which is a numeric type. Sum of products grouped by category for instance. Add the pie visualization type and then email the question to yourself by creating a pulse based on the question. Should also work in slack if you have a token handy.

Alternatively, there are some helper functions to render this for you.

```clojure
dev=> (doto 'dev.render-png require in-ns)
dev.render-png
render-png=> (render-card-to-png 3257)
nil
```

### CODE REVIEW

File: src/metabase/pulse/render/body.clj [hunk 1]
```diff
@@ -242,13 +242,19 @@
...
-        legend-colors               (zipmap (map (comp str x-axis-rowfn) rows) (cycle colors))
+        rows                        (map (juxt (comp str x-axis-rowfn) y-axis-rowfn)
+                                         (common/non-nil-rows x-axis-rowfn y-axis-rowfn rows))
+        legend-colors               (zipmap (map first rows) (cycle colors))
+        total                       (apply + (map second rows))
+        percentages                 (into {}
+                                          (for [[label value] rows]
+                                            [label (if (zero? total)
+                                                     (tru "N/A")
+                                                     (let [f (DecimalFormat. "###,###.##%")]
```
Comment (howonlee) [inline src/metabase/pulse/render/body.clj:253]: is this compatible with our l10n? unless there's cool magic stuff I missed it seems to hardcode American representation (aka, dot as the decimal separator etc etc)?

Comment (dpsutton) [reply src/metabase/pulse/render/body.clj:253]: image
As I understand it from the [javadocs for DecimalFormat](https://docs.oracle.com/javase/7/docs/api/java/text/DecimalFormat.html) the formatting language uses American Locale symbols but there are localized in the implementation.

Comment (dpsutton) [reply src/metabase/pulse/render/body.clj:253]: here's Germany:
```
clj -J-Duser.country=DE
Clojure 1.10.3
user=> (.format (java.text.DecimalFormat. "###,###.##%") 123456789.123455)
"12.345.678.912,35%"
user=>
```

Comment (howonlee) [reply src/metabase/pulse/render/body.clj:253]: hey, magic I didn't know about. cool beans

**Review by howonlee — APPROVED** [review]
it works, I learned a thing about java datetime formatter, no other material comments

**Pull request merged**